codegen/src/main/scala/akka/grpc/gen/scaladsl/ScalaCodeGenerator.scala

```diff
@@ -65,12 +66,16 @@ abstract class ScalaCodeGenerator extends CodeGenerator {
     b.build()
   }
 
+  // flags listed in akkaGrpcCodeGeneratorSettings's description
   private def parseParameters(params: String): GeneratorParams =
     params.split(",").map(_.trim).filter(_.nonEmpty).foldLeft[GeneratorParams](GeneratorParams()) {
-      case (p, "java_conversions")      => p.copy(javaConversions = true)
-      case (p, "flat_package")          => p.copy(flatPackage = true)
-      case (p, "grpc")                  => p.copy(grpc = true)
-      case (p, "single_line_to_string") => p.copy(singleLineToProtoString = true)
-      case (x, _)                       => x
+      case (p, "java_conversions")            => p.copy(javaConversions = true)
+      case (p, "flat_package")                => p.copy(flatPackage = true)
+      case (p, "single_line_to_string")       => p.copy(singleLineToProtoString = true) // for backward-compatibility
+      case (p, "single_line_to_proto_string") => p.copy(singleLineToProtoString = true)
+      case (p, "ascii_format_to_string")      => p.copy(asciiFormatToString = true)
+      case (p, "no_lenses")                   => p.copy(lenses = false)
+      case (p, "retain_source_code_info")     => p.copy(retainSourceCodeInfo = true)
```

Comment:
I'm wondering this parsing can handle the key-value form passing via different plugins? (currently only `gradle`)
e.g. `single_line_to_string=true`. I guess current test infra can't catch this, we only have `serverPowerApis` been used. https://github.com/akka/akka-grpc/blob/f6536403a5cccecb6398ec0485e0ba45f9fa994f/plugin-tester-scala/build.gradle#L34

Reply:
The `key=value` format is only supported for/via: https://github.com/akka/akka-grpc/blob/717731d479b8ad9e9103a6a2e5ac49b813b0ab93/codegen/src/main/scala/akka/grpc/gen/scaladsl/ScalaCodeGenerator.scala#L45.

I have been thinking about the flag parsing when opening this PR, as the current solution clearly does not scale:
1. the akka-grpc specific enum values are declared in the plugin, but can't be accessed from the generators (otherwise we would have a cyclic dependency between the `codegen` and the sbt plugin modules) 
1. unhandled enum values are silently ignored.

I did not have the gradle use case in mind (I had started to push more stuff in the sbt plugin). Maybe we can follow-up on a ticket to brainstorm? My intent here was just a quick one-shot to expose new ScalaPB flags.

Reply:
> The `key=value` format is only supported for/via:
> 
> https://github.com/akka/akka-grpc/blob/717731d479b8ad9e9103a6a2e5ac49b813b0ab93/codegen/src/main/scala/akka/grpc/gen/scaladsl/ScalaCodeGenerator.scala#L45
> 
> .
> I have been thinking about the flag parsing when opening this PR, as the current solution clearly does not scale, but identified a couple of challenges:
> 
> 1. the Akka-gRPC-specific enum values are declared in the plugin, but can't be accessed from the generators (otherwise we would have a cyclic dependency between the `codegen` and the sbt plugin modules)
> 2. The ScalaPB-specific enums can't seem to be listed/retrieved from upstream
> 
> As a result, unhandled enum values are silently ignored.

> I did not have the gradle use case in mind (I had started to push more stuff in the sbt plugin). Maybe we can follow-up on a ticket to brainstorm? 

Yes, we have discussed this at (https://github.com/akka/akka-grpc/pull/645#issuecomment-516171609). I'm going to do a follow-up refactoring for standard generator settings enums in `codegen` module. That would be nice if we have a ticket to track these. The supposed changes including to:

1. Move **generator options** and **parsing** into a single place (in `codegen` module) to be shared across sbt and maven plugins, as well as scaladsl and javadsl code generator.
2. A better way I'm thinking is taking this https://github.com/akka/akka-grpc/issues/451 for a more scalable and unified way to tackle these options, and we can find some way to make the current configuration system compatible.

> My intent here was just a quick one-shot to expose new ScalaPB flags.

Yes, that make sense!